sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonDataSource.scala

```diff
@@ -175,33 +185,43 @@ object MultiLineJsonDataSource extends JsonDataSource {
       .values
   }
 
-  private def createParser(jsonFactory: JsonFactory, record: PortableDataStream): JsonParser = {
+  private def createParser(
+      jsonFactory: JsonFactory,
+      record: PortableDataStream,
+      charset: Option[String] = None): JsonParser = {
     val path = new Path(record.getPath())
     CreateJacksonParser.inputStream(
       jsonFactory,
-      CodecStreams.createInputStreamWithCloseResource(record.getConfiguration, path))
+      CodecStreams.createInputStreamWithCloseResource(record.getConfiguration, path),
+      charset
+    )
   }
 
   override def readFile(
       conf: Configuration,
       file: PartitionedFile,
       parser: JacksonParser,
       schema: StructType): Iterator[InternalRow] = {
+    def createInputStream() = {
```

Comment:
To be honest, I think I wouldn't have a nested function to deduplicate two lines ..

Reply:
I don't know but for me it is easier to read if I see the same name instead of comparing 2 expressions and recognizing them as the same. I believe the below is more complicated:
```
Utils.tryWithResource(CodecStreams.createInputStreamWithCloseResource(conf, new Path(new URI(file.filePath)))) { is =>
        UTF8String.fromBytes(ByteStreams.toByteArray(is))
 }
```

Reply:
I admit it but nested function is a thing to avoid too unless if it really cleans up the codes or is required. Let's just leave it as was if both don't look quite cool .. :). At least that's usually what I do.